stdlib/public/Concurrency/ExecutorAssertions.swift

```diff
@@ -32,9 +32,18 @@ extension SerialExecutor {
   /// * In `-O` builds (the default for Xcode's Release configuration), stops
   ///   program execution.
   ///
-  /// * In `-Ounchecked` builds, the optimizer may assume that this function is
-  ///   never called. Failure to satisfy that assumption is a serious
-  ///   programming error.
+  /// - Note: This check is performed against the actor's serial executor,
+  ///   meaning that / if another actor uses the same serial executor--by using
+  ///   that actor's serial executor as its own ``Actor/unownedExecutor``--this
+  ///   check will succeed , as from a concurrency safety perspective, the
+  ///   serial executor guarantees mutual exclusion of those two actors.
```

Comment:
Trying to fix some odd punctuation:

```suggestion
  /// - Note: Because this check is performed against the actor's serial executor,
  ///   if another actor uses the same serial executor--by using
  ///   that actor's serial executor as its own ``Actor/unownedExecutor``--this
  ///   check will succeed. From a concurrency safety perspective, the
  ///   serial executor guarantees mutual exclusion of those two actors.
```